src/DotVVM.Framework/Controls/HtmlWriter.cs

```diff
@@ -352,6 +366,9 @@ public void RenderEndTag()
                 writer.Write("</");
                 writer.Write(tag);
                 writer.Write(">");
+
+                if (this.enableWarnings && IsSelfClosing(tag))
+                    Warn($"Element {tag} is self closing but contains content. The browser may interpret the start tag as self-closing and put the 'content' into it's parent.");
```

Comment:
Come on, you even have it here (still looking at "self-closing").

~"it's parent"~ -> "its parent".

Reply:
Thanks, fixed that.